### PR TITLE
Implement draw_optimal_set for optset::AbstractOptimalEdgeSet

### PR DESCRIPTION
**Pull request by joshyaffee**
I implemented draw_optimal_set for highlighting a set of edges (previously, only implemented for vertices). 
I did not add docstrings in order to stay consistent with with the other method. I'm happy to add it if requested.
I used the following code to test the implementation:

```julia
g = Graphs.PetersenGraph()
optimal_edges = compute(MaximumMatching, g)
draw_optimal_set(g, optimal_edges)
```

but did not include it in unit tests, since the validity is confirmed by visual inspection.
Another method to consider highlights vertices and edges. in one plot--it would be simple to combine the two methods.

# Pull Request Checklist

Thank you for contributing to `GraphProperties.jl`!

## Documentation and Style

- [ ] I have added docstrings to any new functions or types, as applicable.
- [ x] My code follows the project's coding style and guidelines.

## New Features and Code Changes

- [ ] If you have added a new algorithm or function, you have also included corresponding unit tests.
- [ x] I confirm that any code added is my own work and is not copied from other sources.

## Review Process

- [ x] I have requested a review from at least one other contributor.

## Additional Notes

- Please describe any additional changes or information relevant to this PR.

---

### Developer's Declaration:

By submitting this Pull Request, I confirm the following:

- [ x] The code I am submitting is my own work and I have the rights to use and distribute it.
- [ x] I confirm that the code I have added is not copied from other sources and that I have appropriately cited or acknowledged any external code or references used.
- [x ] I understand that my code will be reviewed and any changes requested must be addressed.

## PR Type

What kind of change does this PR introduce?
- [ ] Bugfix
- [x ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation content changes
- [ ] Unit Testing
- [ ] Other... Please describe:

## What is the new behavior (if this is a feature change)?

- [ x] Yes: Added second method for draw_optimal_set
- [ ] No

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

**Comment by joshyaffee** [comment]
@RandyRDavila As noted in the initial comment, I did not add docstrings in order to stay consistent with with the other method for this function. I'm happy to add it if requested. Also, I suppose I could add a unit test that relies on exception handling to make sure the plot works, but I don't know if there's a good way to add a unit test beyond that for testing the plotting functionality.

File: src/Invariants/drawing/draw_optimal_sets.jl [hunk 1]
```diff
@@ -17,4 +17,28 @@ function draw_optimal_set(
...
         layout=stressmajorize_layout
     )
+end
+
+function draw_optimal_set(
+    g::AbstractGraph,
+    optset::AbstractOptimalEdgeSet
+)
+
+    # initialize array of colors for each edge
```
Comment (RandyRDavila) [inline src/Invariants/drawing/draw_optimal_sets.jl:27]: Let's try to have complete sentences for code comments (I know I might have missed this myself in many places), so this comment should be:

```julia
# Initialize an array of colors for each edge. 
```

Comment (RandyRDavila) [reply src/Invariants/drawing/draw_optimal_sets.jl:27]: The same for the comments below

**Review by RandyRDavila — APPROVED** [review]
Thanks for this PR! 

I made a comment about code comments being complete sentences. I know I didn't follow this rule with my own code, but going forward, I think we both should try to. Other than this one thing, everything LGTM! (looks got to me)

I'll approve this PR! In the future let's fix up our comments and also add docstrings whenever we add a new method to a function. 

After I approve the PR please merge it. It's always best practice to let the owner of the PR to merge their code.

**Pull request merged**